examples/event_consumer.ex

```diff
@@ -1,10 +1,17 @@
 defmodule ExampleSupervisor do
-  def start_link do
-    import Supervisor.Spec
+  use Supervisor
+  
+  def start_link(args) do
+    Supervisor.start_link(__MODULE__, args, name: __MODULE__)
+  end
 
-    children = [ExampleConsumer]
+  @impl true
+  def init(_init_arg) do
+    children = [
+      {ExampleConsumer, []}
```

Comment:
Do you think we should maybe change this line to just be `ExampleConsumer`? IMO it's more idiomatic and could be less confusing for those who are new to supervision trees (since they're the most likely people to use this kind of example).

Reply:
Yes, that makes sense as the consumer doesn't accept any init args anyway. Also ```children = {ExampleConsumer, []}``` is equal to ```children = [ExampleConsumer]```